helix-core/src/comment.rs

```diff
@@ -91,8 +98,8 @@ mod test {
         let text = state.doc.slice(..);
 
         let res = find_line_comment("//", text, 0..3);
-        // (commented = true, skipped = [line 1], min = col 2)
-        assert_eq!(res, (false, vec![1], 2));
+        // (commented = true, to_change = [line 0, line 2], min = col 2, margin = 1)
+        assert_eq!(res, (false, vec![0, 2], 2, 1));
```

Comment:
There's a TODO for a test with no margin after the comment, can you add a case for it?

Reply:
I tried adding a case reusing the `state` from the previous tests, but it seems that the selection must have been altered by the last test, because when I manually set `state.doc` and toggle the comments, the last line isn't affected; it does however work properly if I reset the selection by adding a copy of this line:
```rs
state.selection = Selection::single(0, state.doc.len_chars() - 1);
```

Reply:
Also, why do these tests have `.clone()` in each of these lines? It doesn't seem to be needed.
```rs
state.selection = state.selection.clone().map(transaction.changes());
```

Reply:
I think it used to be needed, that's why

Reply:
But it isn't needed now, so should I remove them?